README.rst

```diff
@@ -1,44 +1,149 @@
+.. figure:: ./coverage.svg
+   :alt: Coverage Status
+
 ============
 Django Canvas OAuth
 ============
 
-Django Canvas OAuth is a Django app that manages OAuth2 Tokens used to make API calls against a Canvas LMS instance.  The OAuth workflow is managed by this library and a CanvasOAuth2Token model is used to store authenticated tokens. Tokens are short-lived, so some logic is introduced at the point of retrieving the the stored token to capture and handle the necessary refresh calls.
+**Django Canvas OAuth** is a Django app that manages OAuth2 Tokens used to make API calls against a Canvas LMS instance.  
 
+The `OAuth workflow`_ is managed by this library and a CanvasOAuth2Token model is used to store authenticated tokens. 
 
-Quick start
------------
+Tokens are short-lived, so some logic is introduced at the point of retrieving the the stored token to capture and handle the necessary refresh calls.
+
+.. _OAuth workflow: https://canvas.instructure.com/doc/api/file.oauth.html
+
+Installation
+------------
+
+Requires python >= 3.6 and Django >= 2.0
+
+.. code-block:: bash
+
+    pip install git+https://github.com/Harvard-University-iCommons/django-canvas-oauth.git#egg=canvas-oauth
+
+
+Quickstart
+----------
 
 1. Add "canvas_oauth" to your INSTALLED_APPS setting like this::
 
+.. code-block:: python
+    
     INSTALLED_APPS = [
-        ...
+        # ...
         'canvas_oauth.apps.CanvasOAuthConfig',
     ]
 
 2. Include the canvas_oauth URLconf in your project urls.py like this::
 
-    url(r'^oauth/', include('canvas_oauth.urls')),
+.. code-block:: python
+
+    path('oauth/', include('canvas_oauth.urls')),
 
 3. Install middleware to begin the oauth2 dance when a token is not
    present and to consume any errors encountered by the library.
 
-   MIDDLEWARE_CLASSES = [
-    ...
-    'canvas_oauth.middleware.OAuthMiddleware'
-   ]
+.. code-block:: python
+    
+    MIDDLEWARE = [
+        # ...
+        'canvas_oauth.middleware.OAuthMiddleware',
+    ]
+
+4. Run ``python manage.py migrate`` to create the canvas_oauth models.
+
+5. Use the ``get_oauth_token`` method from ``canvas_oauth.oauth`` to obtain a 
+   token. This method contains all of the logic to obtain a new token, refresh 
+   an expired one, or return an existing one.
+
+.. code-block:: python
+
+    from canvas_oauth.oauth import get_oauth_token
+    
+    access_token = get_oauth_token(request)
+    #  Make request to the Canvas API using above token.
+
+
+Settings
+---------
+
+Settings should be added to your django settings module (e.g. ``settings.py``).
+
+
+CANVAS_OAUTH_CLIENT_ID:
+    (required) The client id is the integer client id value of your Canvas developer key. 
+
+CANVAS_OAUTH_CLIENT_SECRET:
+    (required) The client secret is the random string (secret) value of your Canvas developer key.
+
+CANVAS_OAUTH_CANVAS_DOMAIN:
+    (required) The domain of your canvas instance (e.g. canvas.instructure.com)
+
+CANVAS_OAUTH_TOKEN_EXPIRATION_BUFFER:
```

Comment:
I agree, this is a bit confusing on first glance, as it's not a standard parameter you will find in the Oauth2 docs. It's intended to force a refresh of the token before it actually expires (e.g. the `expires_in` provided in the access token response), providing a bit of a buffer so you don't acquire a token that is going to expire immediately. Perhaps it would be clearer to refactor this in terms of an absolute expiration instead of an offset/delta. For now, I'll try adding a bit more of an explanation.